etc/travis-scripts

```diff
@@ -0,0 +1,169 @@
+#!/usr/bin/env bash
+#-------------------------------------------------------------------------------
+# (C) British Crown Copyright 2012-8 Met Office.
+#
+# This file is part of Rose, a framework for meteorological suites.
+#
+# Rose is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# Rose is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with Rose. If not, see <http://www.gnu.org/licenses/>.
+#-------------------------------------------------------------------------------
+set -eu
+set -o xtrace  # for ease of debugging travis builds
+shopt -s extglob
+
+
+export PATH="${PWD}/bin:${HOME}/fcm-master/bin:${HOME}/cylc-master/bin:$PATH"
+
+TMP_DOCS_DIR="${TMPDIR:-/var/tmp}/temp-docs-dir"
+DOCUMENTATION_FILES=!(doc|404*|CHANGES.md|_config.yml)
+GITHUB_REPO="https://github.com/${TRAVIS_REPO_SLUG}"
+
+
+install() {
+    args=("$@")
+
+    # package managers
+    sudo apt-get -q update
+    sudo apt-get install -qy python-pip
+
+    # Cylc
+    if grep 'cylc' <<< "${args[@]}"; then
+        # install Cylc
+        wget 'https://github.com/cylc/cylc/archive/master.tar.gz' -O \
```

Comment:
Now we have migrated from the monolithic ``.yml`` file, it might be useful to move some/all of the site link definitions (or at least the bases e.g. ``https://github.com/[cylc/cylc/]``) from which we extract files etc. to the head of the file so that they are immediately explicit? Then again it might be clearer to define them within the functions they are used only... I'll leave it for consideration.

Reply:
`https://github.com/cylc/cylc/` is only used once so I don't think we would have much to gain, of have I misunderstood?

Reply:
There is not much duplication, true, but the links are constants that are fundamental to the scripts, so putting them at the head of the script might be more appropriate & would make it clear where we are sourcing the codebase installations from. There is not much to gain now, admittedly, but may be useful for future reference. We can leave it be, though.